File: a647(investment_expert)/C++/investment_expert.cpp
#include <iostream>
#include <iomanip>

using namespace std;

double round(double n) {
  bool is_nagative = false;
  if (n < 0) {
    is_nagative = true;
    n *= -1;
  }
  double rounded;
  int check = int(n * 1000) % 10;
  if (check >= 5) rounded = (int)(n * 100 + 1) / 100.0f;
  else rounded = (int)(n * 100) / 100.0f;
  if (is_nagative) return -rounded;
  return rounded;
}

int main(int argc, char const *argv[]) {
  ios_base::sync_with_stdio(false);
  cin.tie(0);
  int n;
  while (cin >> n) {
    double m, p, profit_rate;
    for (int i = 0; i < n; i++) {
      cin >> m >> p;
      profit_rate = (p - m) * 100 / m;
      if (profit_rate >= 10 || profit_rate <= -7)
        cout << fixed << setprecision(2) << round(profit_rate) << "% dispose" << '\n';
      else cout << fixed << setprecision(2) << round(profit_rate) << "% keep" << '\n';
    }
  }
  return 0;
}

File: a647(investment_expert)/Python/investment_expert.py
from sys import stdin, stdout

def round(n: float) -> float:
    is_nagative = False
    rounded = 0.0
    if n < 0:
        is_nagative = True
        n *= -1
    check = int(n * 1000 % 10)
    if check >= 5:
        rounded =  int(n * 100 + 1) / 100
    else: rounded = int(n * 100) / 100
    if is_nagative: return -rounded
    return rounded

while True:
    n = stdin.readline().strip('\n')
    if not n: break
    for i in range(int(n)):
        m, p = map(int, stdin.readline().split())
        profit_rate = (p - m) * 100 / m
        if profit_rate >= 10 or profit_rate <= -7:
            stdout.write(f'{round(profit_rate):.2f}% dispose\n')
        else: stdout.write(f'{round(profit_rate):.2f}% keep\n')
